File: services/groupbuy-service/src/main/java/groupbuy_service/product/event/ProductCreatedEvent.java
package groupbuy_service.product.event;

import groupbuy_service.common.event.DomainEvent;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record ProductCreatedEvent(
        String eventId,
        int version,
        Instant occurredAt,
        String productId,
        String name,
        BigDecimal price,
        int initialStock
) implements DomainEvent {
    public static final String TOPIC = "product.created";

    @Override
    public String getTopic() {
        return TOPIC;
    }

    @Override
    public String getKey() {
        return productId;
    }

    public static ProductCreatedEvent of(String productId, String name, BigDecimal price, int initialStock) {
        return new ProductCreatedEvent(
                UUID.randomUUID().toString(),
                1,
                Instant.now(),
                productId,
                name,
                price,
                initialStock
        );
    }
}

File: services/inventory-service/src/main/kotlin/inventory_service/event/ProductCreatedEvent.kt
package inventory_service.event

import java.math.BigDecimal
import java.time.Instant

data class ProductCreatedEvent(
    val eventId: String,
    val version: Int,
    val occurredAt: Instant,
    val productId: String,
    val name: String,
    val price: BigDecimal,
    val initialStock: Int
)
